File: experience-trpg-backend/DTOs/FichaDto.cs
using System;
using System.Collections.Generic;

namespace experience_trpg_backend.DTOs
{

    public class FichaDto
    {
        public int FichaId { get; set; }
        public string? Nome { get; set; }
        public string? Descricao { get; set; }
        public int SistemaId { get; set; }
        public int? MesaId { get; set; }
        public int? ImagemId { get; set; }

        public ImagemDto? Imagem { get; set; }

        public List<AtributoDto> Atributos { get; set; } = new List<AtributoDto>();
        public List<HabilidadeDto> Habilidades { get; set; } = new List<HabilidadeDto>();
        public List<ProficienciaDto> Proficiencias { get; set; } = new List<ProficienciaDto>();
    }
}

File: experience-trpg-frontend/src/app/dtos/ficha.dto.ts
import { ImagemDto } from './imagem.dto';

export interface AtributoDto {
  atributoId: number;
  nome: string;
  valor: number; // Ou float, se você precisar de valores decimais
  fichaId: number;
}

export interface HabilidadeDto {
  habilidadeId: number;
  nome: string;
  descricao: string;
  fichaId: number;
}

export interface ProficienciaDto {
  proficienciaId: number;
  nome: string;
  proficiente: boolean;
  fichaId: number;
}

export interface FichaDto {
    fichaId: number;
    nome: string;
    descricao: string;
    sistemaId: number;
    mesaId: number;
    imagemId: number;

    imagem?: ImagemDto;

    // Adicionar coleções para atributos, proficiências e habilidades
    atributos: AtributoDto[];      // Lista de Atributos
    habilidades: HabilidadeDto[];  // Lista de Habilidades
    proficiencias: ProficienciaDto[]; // Lista de Proficiências
  }